src/core/Akka.Remote.TestKit/Player.cs

```diff
@@ -71,55 +69,50 @@ private class WaitForClientFSMToConnect : UntypedActor
 
             protected override void OnReceive(object message)
             {
-                var fsm = message as IActorRef;
-                if (fsm != null)
+                if (message is IActorRef fsm)
                 {
                     _waiting = Sender;
                     fsm.Tell(new FSMBase.SubscribeTransitionCallBack(Self));
                     return;
                 }
-                var transition = message as FSMBase.Transition<ClientFSM.State>;
-                if (transition != null)
+
+                if (message is FSMBase.Transition<ClientFSM.State> transition)
                 {
-                    if (transition.From == ClientFSM.State.Connecting && transition.To == ClientFSM.State.AwaitDone)
-                        return;
-                    if (transition.From == ClientFSM.State.AwaitDone && transition.To == ClientFSM.State.Connected)
+                    switch (transition.From)
                     {
-                        _waiting.Tell(Done.Instance);
-                        Context.Stop(Self);
-                        return;
+                        case ClientFSM.State.Connecting when transition.To == ClientFSM.State.AwaitDone:
+                            return;
+                        case ClientFSM.State.AwaitDone when transition.To == ClientFSM.State.Connected:
+                            _waiting.Tell(Done.Instance);
+                            Context.Stop(Self);
+                            return;
+                        default:
+                            _waiting.Tell(new Exception("unexpected transition: " + transition));
+                            Context.Stop(Self);
+                            break;
                     }
-                    _waiting.Tell(new Exception("unexpected transition: " + transition));
-                    Context.Stop(Self);
                 }
-                var currentState = message as FSMBase.CurrentState<ClientFSM.State>;
-                if (currentState != null)
-                {
-                    if (currentState.State == ClientFSM.State.Connected)
-                    {
-                        _waiting.Tell(Done.Instance);
-                        Context.Stop(Self);
-                        return;
 
-                    }
-                }
+                if (message is not FSMBase.CurrentState<ClientFSM.State> { State: ClientFSM.State.Connected }) return;
+                _waiting.Tell(Done.Instance);
+                Context.Stop(Self);
             }
         }
 
         /// <summary>
         /// Enter the named barriers, one after the other, in the order given. Will
         /// throw an exception in case of timeouts or other errors.
         /// </summary>
-        public void Enter(string name)
+        public void Enter(RoleName roleName, string name)
```

Comment:
Had to add `RoleName` support to the `Enter` methods for transmitting barrier information to the `BarrierCoordinator`